python/ray/dag/context.py

```diff
@@ -0,0 +1,54 @@
+from dataclasses import dataclass
+import os
+import threading
+from typing import Optional
+from ray.util.annotations import DeveloperAPI
+
+# The context singleton on this process.
+_default_context: "Optional[DAGContext]" = None
+_context_lock = threading.Lock()
+
+# Default buffer size is 100MB.
+DEFAULT_BUFFER_SIZE_BYTES = int(os.environ.get("RAY_DAG_buffer_size_bytes", 100 * 1e6))
+# Default async_max_queue_size is 0, which means no limit.
+DEFAULT_ASYNC_MAX_QUEUE_SIZE = int(os.environ.get("RAY_DAG_async_max_queue_size", 0))
+
+DEFAULT_MAX_BUFFERED_RESULTS = int(os.environ.get("RAY_DAG_max_buffered_results", 1000))
```

Comment:
I think this is a max of 100GB, which is probably too high.

How about setting DAG buffer size to 1MB (now that we can resize the channel, it's not as big of a deal if it's lower), and keep this at 1000MB so we have default capacity of 1GB?

Reply:
Updated https://github.com/ray-project/ray/pull/46268/commits/01d5afd45be946d4024fc77fa27ce8961c04d432